Level 2

SmokySvyat opened the pull request.

SmokySvyat committed:
Изменения:
- Установлен пакет CRA
- Создана файловая структура проекта
- Добавлены компоненты
- Прописаны роуты в арр
- Добавлены шрифты
- Добавлен файл normalize

SmokySvyat committed:
edited readme

SmokySvyat committed:
Добавлено:
- вёрстка header авторизованного пользователя
- вёрстка по роуту  /main:
    - добавлены секции Promo, NavTab, AboutProject, Techs

SmokySvyat committed:
Завершено:
- вёрстка лендинга
-вёрстка карточки фильма
- вёрстка формы поиска

Предстоит реализовать:
- кнопка "ещё" на странице с карточками фильмов

SmokySvyat committed:
Завершено:
- вёрстка страницы профиля
- вёрстка страницы регистрации

SmokySvyat committed:
Завершено:
- вёрстка разрешения 1280px
- вёрстка лэндинга 768px
- реализована навигация по роутам

SmokySvyat committed:
реализовано открывающееся боковое меню

SmokySvyat committed:
Завершено:
- вёрстка всех разрешений

SmokySvyat committed:
Реализовано:
- рендер карточек фильмов по ТЗ

SmokySvyat committed:
в readme добавлено название макета

AliaksandrHrychonak reviewed (commented):
# Резюме по работе:

## Верстка и JSX


### Критерии, влияющие на работоспособность

- [ ] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.
    - в роут сохраненных фильмов необходимо добавить карточки
- [ ] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
    - Роут `/главная` имеет прокрутку по горизонтали на промежутках  `от 854px до 840px, от 782 до 730`
    - Роут `/movies, /saved-movies` имеет прокрутку по горизонтали на промежутках  `от 782 до 730`
- [ ] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
    - роуты регистрации и войти по заданию `/signup, /signin`
    - Ссылка `Статичный сайт` ведёт в пустоту, проверьте публичность репозитория
    - `portfolio-list__item` в двух ссылках нет необходимости, используйте одну
    - Кнопка `Выйти из аккаунта` должна возвращать пользователя на главную страницу
- [ ] <!--5.55--> Именование классов сделаны по БЭМ.
    - проект имеет БЭМ ошибки, удобно проверить валидатором https://nglazov.github.io/bem-validator-page/ , проконтролируйте данный критерий по всем роутам, пример https://disk.yandex.com/i/Kto2CXR6f0adpg
- [ ] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
    - Все фиксации высоты для текстовых элементов и блоков с динамическим (меняющимся) контентом нужно удалить. Высота блоков "строится" из контента + их отступов. Нужно "почистить" ограничения высоты во всем проекте(Допустимо фиксировать минимальную высоту для декоративных блоков)
    - Содержимое на адаптиве не должно выпадать из своих блоков, для удобной проверки добавьте правило `* { outline: 1px solid red; }` в конец `index.css`
    - Контент страницы должен быть центрирован, на промежуточных разрешениях контент не должно терять центрирование.
        - https://disk.yandex.com/i/aOLGjhJL19fyyA https://disk.yandex.com/i/JM6u_ohxyhEqFg
- [ ] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
    - проект имеет ошибки валидации, удобно проверить валидатором https://validator.w3.org/#validate_by_input , проконтролируйте данный критерий по всем роутам, пример https://disk.yandex.com/i/I3cmdXM09fcHyA
    - в роуте `/profile` не хватает сематического тега `<section>`
    - `article_about-project` в данном случае `<article>` не имеет смысла так как сущность невозможно переиспользовать в отрыве от секции
    - Список карточек и поисковой инпут является отдельной секцией
    - `profile__title` является заголовком первого уровня
    - `card__title` является заголовком второго уровня
    - Линии полностью декоративные элементы в проекте, их не нужно размечать сущностьями в разметке, для реализации линий используйте css границы или псевдоэлементы, например `profile-brake`
    - Всем кнопкам в проекте нужен заполненный атрибут `type`, если кнопка **не** отвечает за отправку формы то значение будет `"button"`
- [x] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:
    - [x] <!--3.38--> разметка заключена в (), лишние пустые <div> для этого не используются;
    - [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [x] В проекте есть:
    - [x] <!--3.38--> директория images/ с картинками,
    - [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
    - [x] <!--3.38--> директория fonts/ со шрифтами.
    
### Хорошие практики

- [ ] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
    - Всем интерактивным элементам(ссылки, кнопки, чекбоксы, лого) нужен эффект при наведении и выставленный курсор, проконтролируйте данный критерий для всех интерактивных элементов в проекте
        - нет наведения `header__logo switch card__save logo  cross overlay__link  overlay__btn`
        - нет курсора `profile__btn overlay__btn`
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> Иконки экспортированы в формате SVG.
- [ ] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
    - Всем инпутам проекта нужен стилизованный фокус и переопределённый `outline`
- [ ] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
    - Поисковой инпут нужно сделать обязательным к заполнению
    - Всем полям **Пароль** нужна валидация минимальной/максимальной длинны значения
- [x] <!--2.14--> reset.css запрещен.

### Рекомендации

- [x] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [ ] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
    - `card__img` альт необходимо сделать уникальным(значение следует брать из пропса)
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;
    
## Количество баллов: **42**


### Доп комментарии

- Неверно указан язык документа

SmokySvyat committed:
Исправлено:
 - нэйминг роутов /signup, /signin
 - ссылка на "Статичный сайт"
 - в portfolio-list__item вторая ссылка заменена на span
 - кнопка "Выйти из аккаунта" переводит на "Главную" страницу
 - ошибки БЭМ валидатора
 - ошибки HTML валидатора
 - исправлен тег на семантический в роуте /Profile
 - поиск перенесён в секцию со списком карточек
 - исправлены заголовки карточки и страницы профиля
 - свёрстаны декоративные линии
 - кнопкам прописан атрибут
 - поисковой инпут сделан обязательным
 - добавлена валидация полям "Пароль" по min/max длинне
 - в card__img прописан alt
 - изменён язык документа

SmokySvyat committed:
Исправлено:
  - добавлено наведение и курсор на элементы
  - инпутам задана стиль фокуса, outline сброшен
  - контент отцентрирован
  - убрана фиксация высоты текстовых элементов
  - исправлен адаптив
  - убрана горизонтальная прокрутка
  - в роут сохранённых фильмов добавлены карточки

AliaksandrHrychonak reviewed (commented):
# Резюме по работе:

## Верстка и JSX


### Критерии, влияющие на работоспособность

- [x] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.
- [x] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [x] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
- [ ] <!--5.55--> Именование классов сделаны по БЭМ.
    - проект имеет БЭМ ошибки, удобно проверить валидатором https://nglazov.github.io/bem-validator-page/ , проконтролируйте данный критерий по всем роутам, пример https://disk.yandex.com/i/No19BbBBeJQZGQ
- [x] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
- [ ] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
    - в роуте `/profile` не хватает сематического тега `<main>`
    - `article_about-project` в данном случае `<article>` не имеет смысла так как сущность невозможно переиспользовать в отрыве от секции
    - Список карточек и поисковой инпут является отдельной секцией
     - `portfolio-list__item` иконку необходимо вложить в ссылку
- [x] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:
    - [x] <!--3.38--> разметка заключена в (), лишние пустые <div> для этого не используются;
    - [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [x] В проекте есть:
    - [x] <!--3.38--> директория images/ с картинками,
    - [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
    - [x] <!--3.38--> директория fonts/ со шрифтами.
    
### Хорошие практики

- [ ] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
    - Всем интерактивным элементам(ссылки, кнопки, чекбоксы, лого) нужен эффект при наведении и выставленный курсор, проконтролируйте данный критерий для всех интерактивных элементов в проекте
        - нет наведения  `search__submit-btn `
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> Иконки экспортированы в формате SVG.
- [x] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
- [x] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
- [x] <!--2.14--> reset.css запрещен.

### Рекомендации

- [x] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;
    
## Количество баллов: **89**

SmokySvyat committed:
Исправлено:
 - добавлено наведение на кнопку submita поиска
 - в роуте /profile исправлен тег на семантически верный
 - переименован класс текста в статье о проекте
 - форма поиска и список карточек вынесены в отдельные секции
 - испралена ошибка БЭМ валидатора

SmokySvyat committed:
исправлен тег на семантически верный
